transformer/diff.go

```diff
@@ -2,76 +2,112 @@ package transformer
 
 import (
 	"fmt"
+	"regexp"
+	"strconv"
 	"strings"
 )
 
 type DiffTransformer struct {
-	rawDiff   string
-	fileDiffs []string
-	segments  []string
+	rawDiff         string
+	numberedRawDiff string
+	fileDiffs       map[string]string
+	fileSegments    map[string][]string
 }
 
 const segmentLength = 4000
 
 func (dt *DiffTransformer) Transform(rawDiff string) {
 	dt.rawDiff = rawDiff
-
```

Comment:
Missing Test: It would be helpful to have a unit test for the 'numberRawDiff' function.